pkg/steps/template_test.go

```diff
@@ -0,0 +1,219 @@
+				template := makeTestTemplate()
+				pod := makeTestPod()
+
+				pod.Spec.Containers = append(pod.Spec.Containers, coreapi.Container{
+					Name: "test",
+					Resources: coreapi.ResourceRequirements{
+						Requests: coreapi.ResourceList{
+							"cpu":    *resource.NewQuantity(1, resource.DecimalSI),
+							"memory": *resource.NewQuantity(3*1024*1024*1024, resource.BinarySI)},
+						Limits: coreapi.ResourceList{"memory": *resource.NewQuantity(10*1024*1024*1024, resource.BinarySI)},
+					},
+				})
+
+				injectPodToTemplateHelper(pod, template)
+				return *template
+			},
+			expected: func() templateapi.Template {
+				template := makeTestTemplate()
+				pod := makeTestPod()
+
+				pod.Spec.Containers = append(pod.Spec.Containers, coreapi.Container{
+					Name:      "test",
+					Resources: makeTestResources(),
+				})
+
+				injectPodToTemplateHelper(pod, template)
+				return *template
+			},
+		},
+	}
+
+	fakecs := ciopTestingClient{
+		kubecs:  fake.NewSimpleClientset(),
+		imagecs: nil,
+		t:       t,
+	}
+
+	podClient := NewPodClient(fakecs.Core(), nil, nil)
+	templateFakeCs := templatefake.NewSimpleClientset()
+	templateV1 := templateFakeCs.TemplateV1()
+	tClient := NewTemplateClient(templateV1, templateV1.RESTClient())
+
+	for _, tc := range testCases {
+		t.Run(tc.testID, func(t *testing.T) {
+			templateFakeCs.Fake.PrependReactor("create", "templateinstances", func(action coretesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(coretesting.CreateAction)
+				instance := createAction.GetObject().(*templateapi.TemplateInstance)
+				expectedTemplate := tc.expected()
+
+				if !equality.Semantic.DeepEqual(expectedTemplate, instance.Spec.Template) {
+					t.Fatal(diff.ObjectDiff(expectedTemplate, instance.Spec.Template))
+				}
+
+				// No errors found, skip the test.
+				t.Skip()
+
+				return true, nil, nil
+			})
+
+			template := tc.crippleFunc()
+			templateExecutionStep := TemplateExecutionStep(&template, api.NewDeferredParameters(), podClient, tClient, "", makeTestJobSpec(), makeTestResourceConfiguration())
+			templateExecutionStep.Run(context.Background(), false)
+
+		})
+	}
+}
+
+func makeTestTemplate() *templateapi.Template {
+	return &templateapi.Template{
+		TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+		ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+		Parameters: []templateapi.Parameter{
+			{Name: "IMAGE_FORMAT", Required: false},
+		},
+	}
+}
+
+func makeTestResources() coreapi.ResourceRequirements {
+	return coreapi.ResourceRequirements{
+		Requests: coreapi.ResourceList{
+			"cpu":    *resource.NewQuantity(3, resource.DecimalSI),
+			"memory": *resource.NewQuantity(8*1024*1024*1024, resource.BinarySI)},
+		Limits: coreapi.ResourceList{"memory": *resource.NewQuantity(10*1024*1024*1024, resource.BinarySI)},
+	}
+}
+
+func makeTestResourceConfiguration() api.ResourceConfiguration {
+	return map[string]api.ResourceRequirements{
+		"*": {
```

Comment:
Is this ever used?

Reply:
I wanted to add a case that there are multiple resources in the config. It's not used, and it's not supposed to.